src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocument.cs

```diff
@@ -219,6 +222,96 @@ private void RestructureItem(TocItemViewModel item, List<TocItemViewModel> items
             }
         }
 
+        private void ReportDependency(ImmutableList<FileModel> models, IHostService host, ImmutableDictionary<string, TocItemInfo> tocModelCache, int parallelism)
+        {
+            var nearest = new Dictionary<string, Toc>();
+            models.RunAll(model =>
+            {
+                var wrapper = tocModelCache[model.OriginalFileAndType.FullPath];
+
+                // If the TOC file is referenced by other TOC, not report dependency
+                if (wrapper.IsReferenceToc)
+                {
+                    return;
+                }
+                var outputFile = GetOutputPath(model.FileAndType);
+                var item = wrapper.Content;
```

Comment:
the only case that doesn't work for not-in-toc articles is that its nearest toc file is removed. Nice catch, i'll fix it

---
In reply to: [100463822](https://github.com/dotnet/docfx/pull/1285#discussion_r100463822) [](ancestors = 100463822)